rdtools/test/irradiance_rescale_test.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+from pandas.testing import assert_series_equal
+from rdtools import irradiance_rescale
+from rdtools.normalization import ConvergenceError
+import pytest
+
+
+@pytest.fixture
+def simple_irradiance():
+    times = pd.date_range('2019-06-01 12:00', freq='15T', periods=5)
+    time_series = pd.Series([1,2,3,4,5], index=times)
+    return time_series
+
+
+@pytest.mark.parametrize("method", ['iterative', 'single_opt'])
+def test_rescale(method, simple_irradiance):
+    # test basic functionality
+    modeled = simple_irradiance
+    measured = 1.05 * simple_irradiance
+    rescaled = irradiance_rescale(measured, modeled, method=method)
+    expected = measured
+    assert_series_equal(rescaled, expected, check_exact=False)
+    
+
+def test_warn(simple_irradiance):
+    with pytest.warns(UserWarning,
+                      match="underlying calculations for irradiance_rescale"):
+        _ = irradiance_rescale(simple_irradiance, simple_irradiance)
+
+
+def test_max_iterations(simple_irradiance):
```

Comment:
This raises UnboundLocalError, a subclass of NameError. This works, but curious why you decided to test for NameError?

Reply:
Good catch!  I used NameError because I had been running the body of `irradiance_rescale` manually in a console to explore its behavior -- since it was executing at global scope, it raised NameError instead of UnboundLocalError.